eclipse-collections/src/main/java/org/eclipse/collections/impl/set/mutable/SetAdapter.java

```diff
@@ -272,63 +272,63 @@ public int hashCode()
     @Override
     public MutableBooleanSet collectBoolean(BooleanFunction<? super T> booleanFunction)
     {
-        BooleanHashSet result = new BooleanHashSet();
+        MutableBooleanSet result = BooleanSets.mutable.empty();
         this.forEach(new CollectBooleanProcedure<>(booleanFunction, result));
         return result;
     }
 
     @Override
     public MutableByteSet collectByte(ByteFunction<? super T> byteFunction)
     {
-        ByteHashSet result = new ByteHashSet(this.size());
+        MutableByteSet result = ByteSets.mutable.empty();
```

Comment:
why did you choose to initialize as empty here instead of using the set size?

Reply:
It seemed odd to me that we were pre-sizing any collect methods for sets. It was even more odd for byte, which has at  most 256 values in a Set. I noticed some of the collectByte implementations didn't pre-size. We could pre-size with other types, but since we are collecting, we have no idea how to predict whether the source size will be equal to the target size anyway, so I opted for being consistent with our object collect for UnifiedSet which does not pre-size.

Reply:
Makes sense 👍